modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java

```diff
@@ -396,72 +399,53 @@ public static void trace(DeliverableAndesMetadata metadata, Slot slot, String de
         private static final String IS_DTX = "isDtx: ";
         private static final String SLOT_ID = "slotId: ";
 
+        /**
+         * Prefixes
+         */
         static final String MESSAGE_TRACE = "Message { ";
         static final String ACKNOWLEDGMENT_TRACE = "Ack { ";
         static final String DTX_TRACE = "Dtx { ";
         static final String TX_TRACE = "Transaction { ";
 
-        private final String traceHeader;
+        /**
+         * Field separator
+         */
+        private static final String FIELD_SEPARATOR = " , ";
 
+        /**
+         * String builder for trace message
+         */
         private StringBuilder messageContent;
 
         /**
          * Create a trace builder class
+         *
          * @param traceHeader type of trace message
          */
         TraceBuilder(String traceHeader) {
             this.messageContent = new StringBuilder();
-            this.traceHeader = traceHeader;
-        }
-
-        /**
-         * Check whether the trace header is added
-         * @return true if trace header is already present
-         */
-        private boolean containTraceHeader() {
-            return messageContent.length() != 0;
-        }
-
-        /**
-         * Add the trace header
-         */
-        private void addTraceHeader() {
             messageContent.append(traceHeader);
         }
 
         /**
-         * Check if the trace header is added. If not added, add the
-         * trace header
-         */
-        private void checkAndAddTraceHeader() {
-            if (containTraceHeader()) {
-                messageContent.append(" , ");
-            } else {
-                addTraceHeader();
-            }
-        }
-
-        /**
          * Set message id to the trace
+         *
          * @param id Message id
          * @return TraceBuilder
          */
         TraceBuilder setMessageId(long id) {
-            checkAndAddTraceHeader();
-            messageContent.append(MESSAGE_ID);
-            messageContent.append(id);
+            messageContent.append(FIELD_SEPARATOR).append(MESSAGE_ID).append(id);
```

Comment:
Would this have a trace message like `Message {  , id: 43543534435 }`  if we add the separator without checking?